src/core/exec-invoke.c

```diff
@@ -888,23 +888,26 @@ static int get_fixed_group(
         return 0;
 }
 
-static int get_supplementary_groups(const ExecContext *c, const char *user,
-                                    const char *group, gid_t gid,
-                                    gid_t **supplementary_gids, int *ngids) {
-        int r, k = 0;
-        int ngroups_max;
-        bool keep_groups = false;
-        gid_t *groups = NULL;
-        _cleanup_free_ gid_t *l_gids = NULL;
+static int get_supplementary_groups(
+                const ExecContext *c,
+                const char *user,
+                gid_t gid,
+                gid_t **ret_supplementary_gids,
+                int *ret_ngids) {
```

Comment:
could also make this `size_t`? counts memory objects after all

Reply:
Now, `ret_ngids` is dropped and the number of GIDs is embedded in the return value, to make it consistent with `getgroups_alloc()`.